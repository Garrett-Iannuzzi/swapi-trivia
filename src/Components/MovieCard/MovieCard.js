import React from 'react';
import './MovieCard.scss';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const MovieCard = ({ title, episode_id, release_date, getCharacterData, poster }) => {
  return(
    <article className='article-movie'>
      <h4 className='movie-card-title'>{title}</h4>
      <div className='movie-card-div'>
        <img src={poster} alt='movie poster' className='poster' />
          <div className='info-div'>
            <h4 className='h4-episode'>Episode: {episode_id}</h4>
            <h4 className='h4-release'>Released: {release_date}</h4>
            <Link to={`/movies/:${episode_id}`}>
              <button
                id={episode_id}
                className='btn-characters'
                onClick={ (event) => getCharacterData(event.target.id) }>View Characters</button>
            </Link>
          </div>
      </div>
    </article>
  )
}

export default MovieCard;

MovieCard.propTypes = {
  title: PropTypes.string,
  episode_id: PropTypes.number,
  release_date: PropTypes.string,
  getCharacterData: PropTypes.func,
  poster: PropTypes.string
}