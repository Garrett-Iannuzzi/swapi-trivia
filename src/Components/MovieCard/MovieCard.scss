.article-movie {
  border: solid 3px white;
  border-radius: 10px;
  background-color: black;
  color: white;
  display: flex;
  flex-direction: column;
  height: 340px;
  align-items: center;
  margin: 10px;
  opacity: .9;
  width: 30%;
  min-width: 400px;
    &>.movie-card-title {
      font-size: 1.8em;
      border-bottom: solid 1px;
      margin-top: 10px;
      margin-bottom: 10px;
    }
  }

  .article-movie:hover {
    border: solid 3px indianred;
    opacity: 1;
  }

.poster {
  border: solid 2px white;
  height: 220px;
  width: 165px;
}

.movie-card-div {
  display: flex;
  flex-wrap: wrap;
}

.info-div {
  display: flex;
  flex-direction: column;
  margin: 20px;
}

.h4-episode {
  font-size: 1.7em;
  margin-bottom: 0;
}

.h4-release {
  margin-top: 10px;
  font-size: .8em;
}

.btn-characters {
  width: 100%;
  cursor: pointer;
  color: white;
  background-color: black;
  height: 30px;
  font-size: 1em;
  border-radius: 4%;
}

.btn-characters:hover {
  transform: scale(1.05);
  background-color: white;
  color: black;
}