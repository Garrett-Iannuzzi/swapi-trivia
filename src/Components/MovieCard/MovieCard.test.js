import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from '../MovieCard/MovieCard.js';

describe('MovieCard', () => {

  let wrapper, mockGetCharacterData, mockMovie;

  beforeEach(() => {
    mockGetCharacterData = jest.fn()
    mockMovie = {
        title:'Some Movie Title',
        episode_id: 28,
        release_date:'2020-05-22',
        id: 28
      };
    wrapper = shallow(<MovieCard 
        title={mockMovie.title}
        episode_id={mockMovie.episode_id}
        release_date={mockMovie.release_date}
        id={mockMovie.episode_id} 
        getCharacterData={mockGetCharacterData}
      />)
  });

  it('Should match the snapshot with data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  });


})