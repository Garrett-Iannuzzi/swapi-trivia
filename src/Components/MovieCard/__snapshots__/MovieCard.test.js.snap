// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard Should match the snapshot with data passed in 1`] = `
<article
  className="article-movie"
>
  <h4>
    Some Movie Title
  </h4>
  <h4>
    Episode: 
    28
  </h4>
  <h4>
    Released: 
    2020-05-22
  </h4>
  <Link
    to="/movies/:28"
  >
    <button
      className="btn-characters"
      id={28}
      onClick={[Function]}
    >
      View Characters
    </button>
  </Link>
</article>
`;
